### ISSUE TITLE
Bun crashing while installing packages 

### ISSUE THREAD

**Issue by itsyashvardhan** [oven-sh/bun#12059]
### How can we reproduce the crash?

![image](https://github.com/oven-sh/bun/assets/68675629/1bf92fd7-d1f3-45c5-bf1b-3ce099916e2c)

### Stack Trace (bun.report)

Bun v1.1.14 ([`7c27f3f`](<https://github.com/oven-sh/bun/tree/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f>)) on windows x86_64 [InstallCommand]

**panic**: index out of bounds

- [`install.zig:4885`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/install/install.zig#L4885>): `enqueueDependencyWithMain`
- [`install.zig:5772`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/install/install.zig#L5772>): `processDependencyList`
- [`install.zig:6595`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/install/install.zig#L6595>): `runTasks`
- [`install.zig:12707`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/install/install.zig#L12707>): `installPackages`
- [`install.zig:13531`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/install/install.zig#L13531>): `installWithManager`
- [`filter_run.zig:431`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/cli/filter_run.zig#L431>): `install`
- [`cli.zig:1466`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/cli.zig#L1466>): `start`
- [`cli.zig:62`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/cli.zig#L62>): `start`
- [`main.zig:51`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/main.zig#L51>): `main`
- [`exe_common.inl:78`](<https://github.com/oven-sh/bun/blob/7c27f3f9b43eff454d2a48b64c5d083a5ae49e9f/src/src/startup/exe_common.inl#L78>): `invoke_main`

**Comment by PythonicBoat** [comment]
On trying to update bun , I have already tried reinstalling bun on my system after performing a clean uninstall

![image](https://github.com/oven-sh/bun/assets/68675629/e220a9be-a326-43b6-8115-84be7b6b7b34)

**Comment by PythonicBoat** [comment]
any updates on this @dylan-conway

**Comment by dylan-conway** [comment]
Could you share the package.json and bun.lockb that gives you the `index out of bounds` error? I almost have a fix for `bin executable does not exist on disk`, but I'm struggling to repro the `index out of bounds` bug

**Comment by PythonicBoat** [comment]
sure, 
### package.json 

```json
{
  "name": "xyzproject",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev --turbo",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@radix-ui/react-icons": "^1.3.0",
    "@react-three/drei": "^9.107.1",
    "@react-three/fiber": "^8.16.8",
    "@vercel/analytics": "^1.3.1",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.1.1",
    "framer-motion": "^11.2.12",
    "lucide-react": "^0.397.0",
    "next": "^14.2.4",
    "next-auth": "^4.24.7",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "tailwind-merge": "^2.3.0",
    "tailwindcss-animate": "^1.0.7",
    "three": "^0.165.0",
    "three-globe": "^2.31.1"
  },
  "devDependencies": {
    "typescript": "^5.5.2",
    "@types/node": "^20.14.9",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "postcss": "^8.4.38",
    "tailwindcss": "^3.4.4",
    "eslint": "^9.5.0",
    "eslint-config-next": "14.2.4"
  }
}
```
### bun.lockb file
https://mega.nz/file/zmIwmLaZ#oD2rxQmpYASoDeQyIdrgzyQcnjkshlyC5L9X-d-hc3Y

**Comment by PythonicBoat** [comment]
is this fixed @dylan-conway ?
thanks!

**Comment by dylan-conway** [comment]
Fixed with #12485 in bun v1.1.20
